components/system-api/src/private.rs

```diff
@@ -206,59 +171,80 @@
 //! # Ok(())
 //! # }
 //! ```
+// limitations under the License.
 
-use exonum::{blockchain::ApiSender, crypto::PublicKey, runtime::InstanceId};
+use exonum::{
+    blockchain::{ApiSender, Blockchain, Schema},
+    crypto::PublicKey,
+    helpers::{exonum_version, os_info, rust_version},
+};
 use exonum_api::{self as api, ApiBackend, ApiScope};
 use exonum_node::{ConnectInfo, ExternalMessage, SharedNodeState};
 use futures::Future;
 use serde::{Deserialize, Serialize};
+use std::{sync::Arc, time::SystemTime};
 
-use std::{collections::HashMap, net::SocketAddr, sync::Arc};
-
-/// Short information about the service.
-#[derive(Serialize, Deserialize, Clone, Debug, PartialEq)]
-pub struct ServiceInfo {
-    /// Service name.
-    pub name: String,
-    /// Service identifier for the database schema and service messages.
-    pub id: InstanceId,
+/// Information about the current state of the node.
+#[derive(Debug, Clone, Copy, Serialize, Deserialize, PartialEq)]
+pub struct StatsInfo {
+    /// Height of the last committed block.
+    pub height: u64,
+    /// Total number of uncommitted transactions stored in persistent pool.
+    pub tx_pool_size: u64,
+    /// Total number of transactions in the blockchain.
+    pub tx_count: u64,
+    /// Size of the transaction cache.
+    pub tx_cache_size: usize,
+    /// Work duration of the node.
+    pub uptime: u64,
 }
 
-/// Short information about the current node.
-#[derive(Serialize, Deserialize, Clone, Debug, PartialEq)]
-pub struct NodeInfo {
-    /// Version of the `exonum` crate.
-    pub core_version: Option<String>,
+/// Information about whether it is possible to achieve the consensus between
```

Comment:
Off-topic: Is this information garbage if the node is an auditor? Also, "Consensus enabled and the node has enough connected peers" does **not** mean that consensus can necessarily advance; for example, some of peers may have consensus disabled.

Reply:
Yeah. I'll rephrase this description.